modules/core/src/bitgo.ts

```diff
@@ -427,848 +699,920 @@ const BitGo = function(params) {
-    } else {
-      // check the presence of an encrypted ECDH xprv
-      // if not present, legacy account
-      const encryptedXprv = body.encryptedECDHXprv;
-      if (!encryptedXprv) {
-        throw new Error('Keychain needs encryptedXprv property');
+        const responseDetails = self.handleTokenIssuance(response.body, password);
+        self._token = responseDetails.token;
+        self._ecdhXprv = responseDetails.ecdhXprv;
+
+        // verify the response's authenticity
+        request.verifyResponse(response);
+
+        // add the remaining component for easier access
+        response.body.access_token = self._token;
       }
 
-      const responseDetails = self.handleTokenIssuance(response.body, password);
-      self._token = responseDetails.token;
-      self._ecdhXprv = responseDetails.ecdhXprv;
+      return response;
+    }).call(this)
+      .then(handleResponseResult(), handleResponseError)
+      .nodeify(callback);
+  };
 
-      // verify the response's authenticity
-      request.verifyResponse(response);
+  /**
+   * @param params
+   * - operatingSystem: one of ios, android
+   * - pushToken: hex-formatted token for the respective native push notification service
+   * @param callback
+   * @returns {*}
+   * @deprecated
+   */
+  registerPushToken(params, callback?: NodeCallback<any>): Bluebird<any> {
+    params = params || {};
+    common.validateParams(params, ['pushToken', 'operatingSystem'], [], callback);
 
-      // add the remaining component for easier access
-      response.body.access_token = self._token;
+    if (!this._token) {
+      // this device has to be registered to an extensible session
+      return this.reject('not logged in', callback);
     }
 
-    return response;
-  })
-  .then(handleResponseResult(), handleResponseError)
-  .nodeify(callback);
-};
-
-/**
- *
- * @param params
- * - operatingSystem: one of ios, android
- * - pushToken: hex-formatted token for the respective native push notification service
- * @param callback
- * @returns {*}
- */
-BitGo.prototype.registerPushToken = function(params, callback) {
-  params = params || {};
-  common.validateParams(params, ['pushToken', 'operatingSystem'], [], callback);
+    const postParams = _.pick(params, ['pushToken', 'operatingSystem']);
 
-  if (!this._token) {
-    // this device has to be registered to an extensible session
-    return this.reject('not logged in', callback);
+    return this.post(this.url('/devices'))
+      .send(postParams)
+      .result()
+      .nodeify(callback);
   }
 
-  const postParams = _.pick(params, ['pushToken', 'operatingSystem']);
 
-  return this.post(this.url('/devices'))
-  .send(postParams)
-  .result()
-  .nodeify(callback);
-};
+  /**
+   *
+   * @param params
+   * - pushVerificationToken: the token received via push notification to confirm the device's mobility
+   * @param callback
+   * @deprecated
+   */
+  verifyPushToken(params, callback?: NodeCallback<any>): Bluebird<any> {
```

Comment:
no type on ```params``` here, is that deliberate?

Reply:
Yep, missed a couple, good catch. Will update.